vowpalwabbit/parse_args.cc

```diff
@@ -1672,9 +1674,19 @@ void free_args(int argc, char* argv[])
 }
 
 vw* initialize(
-    options_i& options, io_buf* model, bool skipModelLoad, trace_message_t trace_listener, void* trace_context)
+    config::options_i& options, io_buf* model, bool skipModelLoad, trace_message_t trace_listener, void* trace_context)
 {
-  vw& all = parse_args(options, trace_listener, trace_context);
+  std::unique_ptr<options_i> opts;
```

Comment:
Since the options object is managed externally we can add a custom deleter which is a noop for the unique_ptr, rather than needing to manage the external state of `should_release_options`

Reply:
ah good point